optuna/integration/shap.py

```diff
@@ -0,0 +1,86 @@
+from collections import OrderedDict
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import numpy as np
+
+from optuna._imports import try_import
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.importance._mean_decrease_impurity import MeanDecreaseImpurityImportanceEvaluator
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+with try_import() as _imports:
+    from shap import TreeExplainer
+
+
+class ShapleyImportanceEvaluator(BaseImportanceEvaluator):
+    """Shapley (SHAP) parameter importance evaluator.
+
+    This evaluator fits a random forest that predicts objective values given hyperparameter
+    configurations. Feature importances are then computed as the mean absolute SHAP values.
+
+    .. note::
+
+        This evaluator requires the `sklean <https://scikit-learn.org/stable/>`_ Python package and
+        `SHAP <https://shap.readthedocs.io/en/latest/index.html>`_.
```

Comment:
Since users tend to the stable version, how about referring to the stable docs.
```suggestion
        `SHAP <https://shap.readthedocs.io/en/stable/index.html>`_.
```